### PR TITLE
Gst Credit (new)

### PR DESCRIPTION
**Pull request by alex-rand**
Supersedes #85 to get around merge conflicts.

### CODE REVIEW

**Review by MaxGhenis — COMMENTED** [review]
Thanks Alex - could you please
1. Take a look at my recent changes and check they make sense
2. Add unit tests for all variables with formulas

Then we'll be good to merge

**Comment by MaxGhenis** [comment]
I changed the logic for `gst_credit_category` to match my understanding: the credit treats eldest children in single-parent households as spouses. That wasn't the previous logic. The change broke the test, so IIUC the test needs revision.

Breaking up the current integration test into unit tests for each formula will clarify.

**Comment by alex-rand** [comment]
> I changed the logic for `gst_credit_category` to match my understanding: the credit treats eldest children in single-parent households as spouses. That wasn't the previous logic. The change broke the test, so IIUC the test needs revision.
> 
> Breaking up the current integration test into unit tests for each formula will clarify.

Oops yes your change is correct, thanks

File: policyengine_canada/tests/gov/cra/tax/income/credits/gst_credit/gst_credit_base.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+- name: Person amounts for family of 3
```
Comment (MaxGhenis) [inline policyengine_canada/tests/gov/cra/tax/income/credits/gst_credit/gst_credit_base.yaml:1]: We don't need a unit test for variables without formulas, like this which just uses `adds`, so you can delete this.

File: policyengine_canada/variables/gov/cra/tax/income/credits/gst_credit/gst_credit_reduction.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from policyengine_canada.model_api import *
+
+
+class gst_credit_reduction(Variable):
+    value_type = float
+    entity = Household
+    label = "GST Credit reduction based on income"
+    unit = CAD
+    documentation = " "
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/cra/tax/income/credits/gst_credit/gst_credit_reduction.py:9]:
```suggestion
```

File: policyengine_canada/tests/gov/cra/tax/income/credits/gst_credit/gst_credit_singles_boost.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+- name: Single person with income 15,000 gets a boost of (15,000 - 9919) * 0.02 = 101.62
+  period: 2023
+  input:
+    gst_credit_category: HEAD
```
Comment (MaxGhenis) [inline policyengine_canada/tests/gov/cra/tax/income/credits/gst_credit/gst_credit_singles_boost.yaml:4]: Also below - unit tests should only have `is_married`, `gst_credit_single_parent_household`, and `household_net_income` as inputs
```suggestion
```

File: policyengine_canada/tests/gov/cra/tax/income/credits/gst_credit/is_child_for_gst_credit.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+- name: Family with two children for GST credit based on age and dependent status
```
Comment (MaxGhenis) [inline policyengine_canada/tests/gov/cra/tax/income/credits/gst_credit/is_child_for_gst_credit.yaml:1]: just `is_dependents` and `age` as inputs

File: policyengine_canada/tests/gov/cra/tax/income/credits/gst_credit/is_eldest_child_in_single_household_for_gst_credit.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+- name: Base amounts for family of 3
+  period: 2023
+  input:
```
Comment (MaxGhenis) [inline policyengine_canada/tests/gov/cra/tax/income/credits/gst_credit/is_eldest_child_in_single_household_for_gst_credit.yaml:3]: just `person_index`, `is_child_for_gst_credit`, and `is_married` as inputs

`age` logic should be covered in the `person_index` unit test

**Review by MaxGhenis — COMMENTED** [review]
thanks alex - should be good after unit tests are unitified

File: policyengine_canada/tests/gov/cra/tax/income/credits/gst_credit/gst_credit_person.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+- name: Person amounts for family of 3
+  period: 2023
+  input:
+    people:
```
[the remark this answers is not in the document]
Comment (alex-rand) [reply policyengine_canada/tests/gov/cra/tax/income/credits/gst_credit/gst_credit_person.yaml:4]: ohh I get it now

**Comment by MaxGhenis** [comment]
Dare I say it: git pull upstream master will get this ready to merge

**Comment by MaxGhenis** [comment]
Mystery solved: `git pull upstream master` wasn't getting properly applied because the commit message opened in vim. Exiting vim via `:q` fixed it.

**Pull request merged**